plugins/pulp_python/plugins/models.py

```diff
@@ -1,120 +1,150 @@
+    semantics of PEP 426. https://www.python.org/dev/peps/pep-0426/#id14
+
+    Package - An individual, installable Python package. Uniqueness of a package is determined by
+              the project name, target architecture, and Python version. Note that there may not be
+              a target architecture in the case of architecture agnostic packages. A Python version
+              is also optional if there are no requirements for a specific runtime version. There
+              are two accepted package types, sdist and wheel.
+    Release - A snapshot of the project. A particular release of the project will all share the
+              same version number, but there may be multiple packages for a given release for
+              different architectures, Python versions, and package types.
+    Project - The entire group of packages of all releases.
+
+    The way that PyPI stores the metadata on their end leaves this model open to potential
+    inaccuracies because they assume that all packages of the same project share certain metadata,
+    including mutable fields. In the situation where a mutable field changes between releases, only
+    the latest version is present in the metadata accessible over the API. This is could be
+    particularly problematic for a field like "license". For this reason, most of these fields have
+    been left out of this model. The only fields that are included here are necessary for sync
+    (name, version, filename, path) or are useful for disambiguation between similarly named
+    projects (author, summary). By keeping the scope of the metadata as small as is reasonable, we
+    have less complexity and reduced risk of innacuracy.
+
+    :ivar author: primary author of the package
+    :type author: basestring
+    :ivar filename: name of the file containing the package and metadata
+    :type filename: basestring
+    :ivar md5_digest: md5 checksum provided by PyPI to ensure we got the correct bits
+    :type md5_digest: basestring
+    :ivar name: name of the project, ex. scipy
+    :type name: basestring
+    :ivar packagetype: format of python package, ex bdist_wheel, sdist
+    :type packagetype: basestring
+    :ivar path: relative path to the bits for this package
+    :type path: basestring
+    :ivar summary: one line summary of what the package does
+    :type summary: basestring
+    :ivar version: Contains the distribution's version number. This field must be in the format
+                   specified in PEP 386.
+    :type version: basestring
+    """
 
     author = StringField()
-    author_email = StringField()
-    description = StringField()
-    home_page = StringField()
-    license = StringField()
-    platform = StringField()
+    filename = StringField(required=True)
+    md5_digest = StringField()
+    name = StringField(required=True)
+    packagetype = StringField()
+    path = StringField()
     summary = StringField()
+    version = StringField(required=True)
```

Comment:
Digging into `pkg_resources.parse_version` shows that if the version is valid, it is used. If it is invalid, it is a legacy version and is still supported. In warehouse, new data will be validated, legacy data will be grandfathered in.

My stance is that if it works on PyPI, we should support it, even if it does not follow the PEP. For that reason, I am going to leave validation of version out, and will also leave a comment in the code explaining why.